### PR TITLE
Interpolation time series

### PR DESCRIPTION
**Pull request by garypolzin**
# Description
This isn't incorporated into the pipeline yet but added interpolation alongside extrapolation.
Moved get_selected_value into utils function.
Slimmed down test data for combined extrapolation and interpolation

# How to test
Test data behaving as expected

[Trello](https://trello.com/c/YyEgLyDP/722-interpolation-following-general-pattern-of-data-rather-than-straight-line)

# Developer checklist
- [X] Unit test
- [X] Linked to Trello ticket
- [X] Documentation up to date

### CODE REVIEW

File: tests/unit/test_extrapolation_and_interpolation.py [hunk 1]
```diff
@@ -54,29 +51,34 @@ def test_model_extrapolation_and_interpolation_runs(
...
 
     def test_model_extrapolation_and_interpolation_returns_same_number_of_rows(self):
         self.assertEqual(
             self.extrapolation_and_interpolation_df.count(), self.returned_df.count()
         )
 
     def test_model_extrapolation_and_interpolation_returns_new_columns(self):
-        extrapolation_model_column_name = "extrapolation_" + self.model_column_name
-        # interpolation_model_column_name = "interpolation_" + self.model_column_name
+        extrapolation_model_column_name = "extrapolation_null_values_column_trend_model"
```
Comment (MarionJHolloway) [inline tests/unit/test_extrapolation_and_interpolation.py:62]: You could abstract variables like this to the test_file_data script. They are technically test data for this function, even though they aren't dataframes. THey also don't take up much space, so I don't have an issue with the being here.

**Review by MarionJHolloway — APPROVED** [review]
Couple of small comments. I've approved so I don't slow you down today.

File: tests/unit/test_interpolation_new.py [hunk 2]
```diff
@@ -0,0 +1,236 @@
...
+        self.returned_df = job.model_interpolation(
+            self.interpolation_df,
+            IndCqc.ascwds_filled_posts_dedup_clean,
+            self.interpolation_model_column_name,
+        )
+
+    def test_model_interpolation_row_count_unchanged(self):
+        self.assertEqual(self.returned_df.count(), self.interpolation_df.count())
+
+    def test_model_interpolation_and_interpolation_returns_new_columns(self):
```
[the remark this answers is not in the document]
Comment (garypolzin) [reply tests/unit/test_interpolation_new.py:44]: name of this test is whacky too!

File: tests/unit/test_interpolation_new.py [hunk 3]
```diff
@@ -0,0 +1,236 @@
...
+        )
+        self.expected_df = self.spark.createDataFrame(
+            Data.expected_calculate_residual_returns_none_when_extrapolation_forwards_is_none_rows,
+            Schemas.expected_calculate_residual_schema,
+        )
+
+    def test_calculate_residuals_returns_expected_columns(self):
+        self.assertTrue(self.returned_df.columns, self.expected_df.columns)
+
+    def test_combine_interpolation_returns_none_when_extrapolation_forwards_is_none(
```
[the remark this answers is not in the document]
Comment (garypolzin) [reply tests/unit/test_interpolation_new.py:86]: oh yea, copied the tests but missed the headings!

File: tests/unit/test_interpolation_new.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (MarionJHolloway) [reply tests/unit/test_interpolation_new.py:44]: I didn't even spot that, just the s!

**Pull request merged**